src/components/views/dialogs/ExportDialog.tsx

```diff
@@ -39,18 +39,74 @@ import { useStateCallback } from "../../../hooks/useStateCallback";
 import Exporter from "../../../utils/exportUtils/Exporter";
 import Spinner from "../elements/Spinner";
 import InfoDialog from "./InfoDialog";
+import ChatExport from "../../../customisations/ChatExport";
 
 interface IProps extends IDialogProps {
     room: Room;
 }
 
+const validateNumberInRange = (min: number, max: number) => (value?: string | number) => {
+    const parsedSize = parseInt(value as string, 10);
+    return !(isNaN(parsedSize) || min > parsedSize || parsedSize > max);
+};
+
+interface ExportConfig {
+    exportFormat: ExportFormat;
+    exportType: ExportType;
+    numberOfMessages: number;
+    sizeLimit: number;
+    includeAttachments: boolean;
+    setExportFormat?: Dispatch<SetStateAction<ExportFormat>>;
+    setExportType?: Dispatch<SetStateAction<ExportType>>;
+    setAttachments?: Dispatch<SetStateAction<boolean>>;
+    setNumberOfMessages?: Dispatch<SetStateAction<number>>;
+    setSizeLimit?: Dispatch<SetStateAction<number>>;
+}
+
+/**
+ * Set up form state using "forceRoomExportParameters" or defaults
+ * Form fields configured in ForceRoomExportParameters are not allowed to be edited
+ * Only return change handlers for editable values
+ */
+const useExportFormState = (): ExportConfig => {
+    const config = ChatExport.getForceChatExportParameters();
+
+    const [exportFormat, setExportFormat] = useState(config.format || ExportFormat.Html);
+    const [exportType, setExportType] = useState(config.range || ExportType.Timeline);
+    const [includeAttachments, setAttachments] =
+        useState(config.includeAttachments !== undefined && config.includeAttachments);
+    const [numberOfMessages, setNumberOfMessages] = useState<number>(config.numberOfMessages || 100);
+    const [sizeLimit, setSizeLimit] = useState<number | null>(config.sizeMb || 8);
```

Comment:
`??` instead of `||` here please

Reply:
What's the benefit?

Reply:
If `sizeMb` was set to `0` it'd consistently be respected, vs with the current approach `-1` stays as `-1`, `0` goes to `0`